### DISCUSSION TITLE
5차 데모데이 준비

### DISCUSSION THREAD

**Discussion by Hain-tain** [woowacourse-teams/2024-code-zap#618] [결정 사항]
1. 각자 레벨 4에서 얼마나 팀 프로젝트에 시간을 할애할 수 있나요? (되도록 솔직하고 정확한 수치로 적어주세요.)
2. 5차 데모데이때 어디까지 구현하고 싶은가요?

**Comment by zeus6768** [comment]
### 제우스
1. 일 120분 (회의 제외)
2. 우테코 요구사항 충족 + 유저 피드백 반영 도메인 1개

**Comment by vi-wolhwa** [comment]
1. 투자 가능 시간 (수업X 미션(마감일근처x) 인 날을 가정)
-> 테코톡 전 : 2시간
-> 테코톡 후 : 3~4시간

2. 5차 데모데이 때 구현할 사항
-> 우테코 팀 미션
-> 사소한 QA 건 지속적으로 개선
-> 팀원들의 의견에 따르겠습니다

**Comment by HoeSeong123** [comment]
### 초롱
1. 하루 3시간(회의 포함)
2. 5차 데모데이 요구사항(쿼리 개선), 사용자 피드백 최소 1개 반영(개인적으로 원하는 건 좋아요 기능)

**Comment by kyum-q** [comment]
### 켬미
1. 하루 2~4시간
2. 5차 데모데이 요구사항, [디스커션](https://github.com/woowacourse-teams/2024-code-zap/discussions/577#discussioncomment-10539802)에 있는 리펙터링 기능 중 선별해서 구현

**Comment by jminkkk** [comment]
1. 3-4시간
2. 5차 요구사항 +  https://github.com/woowacourse-teams/2024-code-zap/discussions/577#discussioncomment-10539802 중 테스트 코드, 도메인간 의존성 재설계, 테이블 구조 개선

**Comment by Jaymyong66** [comment]
1. 정확한 수치로 보자면 일 2~3시간이 될 것 같아요. (집중시간)
    다만 유동적으로 가져갈 것 같습니다 ~
2. 전체적으로 보면 데모데이 요구사항 + 우리가 정한 기능 목록과 리팩토링 + UX 피드백 반영 인데요. 
    5차 데모데이까지는 일전에 정한 리스트에서 합의된 기능과 리팩토링을 진행할 것 같습니다.

**Comment by Hain-tain** [comment]
1. 하루에 2-3시간 정도! 
2. 리펙토링 일부 + 각자 분야에서의 5차 데모데이 요구사항 + 좋아요와 관련된 기능

**Comment by zangsu** [comment]
1. 하루 3 시간 (바쁜 날은 일주일에 하루정도 전부 투자할 수 있을 것 같습니다.)
2. 요구사항 반영 + 새로운 기능 개발 하나정도는 있어도 좋을 것 같아요.

**Reply by zangsu** [reply]
OAuth 도입을 고려하는 것도 좋을 것 같네용

**Comment by HoeSeong123** [comment]
## 할 것 목록
### 새로운 기능 (20일 마감)
#### 다른 사람 것을 내가 잘 볼 수 있게 하는 기능
- "구경가기" 좋아요 순 정렬
- "구경가기" 검색
- 좋아요 기능

---

### 백엔드 요구사항
- 쿼리 개선
- 검색 기능 개선

### 백엔드 리팩토링
#### 테스트 코드
- 각 기능에 따른 인수 테스트 구현
- 사용자 시나리오에 따른 통합 테스트 (Http 요청 ~ Http 응답) (E2E 테스트)
- 단위 테스트 작성 (레포지토리, 도메인 등등)
- 현재 작성되어 있지 않은 기능 또는 예외에 대한 테스트 코드 보완
- 현재 작성되어 있는 테스트 코드가 유효한 테스트인지 확인 후 개선
- 테스트 환경에서 사용할 디비 결정 (ex. 페이크디비)
- 인수 테스트 환경에서 발생할 환경상 문제 (ex.시간, 환경 일치) 개선

#### 도메인
각 도메인간 의존성 재설계

---

### 프론트 요구사항
- 목표했던 타겟 환경과 디바이스에서 서비스 핵심 기능 동작 확인
- 성능 리포트 작성 및 리포트 기준으로 필요한 만큼의 성능 최적화 작업 진행
  - 구체적인 성능 개선 대상과 선정한 이유, 목표 지표를 설정한 이유를 반드시 포함한다

### 프론트 리팩토링

**Comment by HoeSeong123** [comment]
# 공지
데일리 스크럼은 매일 10시 5분 ~ 15분. **무조건**
